### PR TITLE
Align balances to intended usage

### LINKED ISSUE #284: Align `Balance` schema, protocol `Resource` example, and HTTP response example
The `Balance` resource is demonstrating a couple variations throughout the spec, prob good to align these.

The variations are:
- `{ data: [ { metadata, data: { balances: [ balanceDetail0, balanceDetail1, balanceDetail2, ... ] }, signature } ] }` (`Balance` as an array within a `Resource`)
- `{ data: [ balanceDetail0, balanceDetail1, balanceDetail2, ... ] }` (`Balance` as the entire resource, but not a `Resource`)

More info here:
- https://github.com/TBD54566975/tbdex/pull/221/files#r1532988686
- https://github.com/TBD54566975/tbdex/pull/221/files#r1533031510

### PR DESCRIPTION
**Pull request by phoebe-lew**
closes #284

### CODE REVIEW

**Review by mistermoe — APPROVED** [review]

**Review by jiyoontbd — APPROVED** [review]

File: specs/http-api/README.md [hunk 1]
```diff
@@ -486,26 +486,10 @@ True
 ### Response
 | Status             | Body                                   |
 | ------------------ | -------------------------------------- |
-| `200: OK.     `    | `{ data: { Balance[] } }` |
+| `200: OK.     `    | `{ data: { Balance[] } }` See [Balance spec](https://github.com/TBD54566975/tbdex/blob/main/specs/protocol/README.md#balance) for the full schema of a Balance resource|
```
Comment (KendallWeihe) [inline specs/http-api/README.md:489]:
```suggestion
| `200: OK.     `    | `{ data: Balance[] }` See [Balance spec](https://github.com/TBD54566975/tbdex/blob/main/specs/protocol/README.md#balance) for the full schema of a Balance resource|
```

**Review by KendallWeihe — APPROVED** [review]
one comment but ✅

**Review by decentralgabe — APPROVED** [review]

File: specs/protocol/README.md [hunk 2]
```diff
@@ -306,18 +306,10 @@ A `Balance` is a protected resource used to communicate the amounts of each curr
...
-        "currencyCode": "USD", 
-        /** same format used to represent currency values across messages */
-        "available": "400.00",
-      },
-      {
-        "currencyCode": "MXN", 
-        "available": "100.00",
-      }
-    ]
+    /** ISO 4217 currency code or widely adopted cryptocurrency code as string */
```
Comment (jiyoontbd) [inline specs/protocol/README.md:309]: with this change, does it mean Alice can only ever have 1 currency balance stored with a PFI?

Comment (jiyoontbd) [reply specs/protocol/README.md:309]: the above spec change says `data: Balance[]` - so should this example show an array of balance details?

**Comment by phoebe-lew** [comment]
Oooh thanks for the reminder @kirahsapong  😄

File: specs/protocol/README.md (hunk 2 above)
Comment (phoebe-lew) [reply specs/protocol/README.md:309]: This is defining what a single balance object looks like. So each balance object has:
```
{
  metadata: {}
  data: {
     "currencyCode": "USD",
     "available": "400.00"
  }
  signature: "abc123"
}
```

Alice can have multiple associated balances.

Comment (jiyoontbd) [reply specs/protocol/README.md:309]: can you write the example to be the balance message I'd get back if I had multiple balances with a PFI?

**Pull request merged**